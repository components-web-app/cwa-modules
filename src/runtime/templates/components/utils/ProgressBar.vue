<template>
  <Transition
    enter-from-class="cwa-transform cwa-opacity-0 cwa-w-0"
    enter-active-class="cwa-duration-200 cwa-ease-out"
    enter-to-class="cwa-opacity-100"
    leave-from-class="cwa-opacity-100"
    leave-active-class="cwa-duration-300 cwa-ease-in cwa-delay-200"
    leave-to-class="cwa-transform cwa-opacity-0"
  >
    <div v-show="props.show" class="cwa-w-full cwa-h-1">
      <div class="cwa-bg-green-400 cwa-h-1 cwa-transition-all cwa-duration-300" :style=" { width: `${props.percent}%` }" />
    </div>
  </Transition>
</template>

<script setup>
const props = defineProps({
  show: { required: true, type: Boolean },
  percent: { required: true, type: Number }
})
</script>
