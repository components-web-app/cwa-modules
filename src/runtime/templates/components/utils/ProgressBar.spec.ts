// @vitest-environment happy-dom
import { describe, expect, test } from 'vitest'
import { shallowMount } from '@vue/test-utils'
import ProgressBar from './ProgressBar.vue'

function createWrapper (loading: boolean, percent: number) {
  return shallowMount(ProgressBar, {
    props: {
      show: loading,
      percent
    }
  })
}

describe('ProgressBar', () => {
  describe('snapshot', () => {
    test('should match snapshot IF bar is shown with specified percentage', () => {
      const wrapper = createWrapper(true, 75)

      expect(wrapper.element).toMatchSnapshot()
    })

    test('should match snapshot IF bar is NOT shown', () => {
      const wrapper = createWrapper(false, 0)

      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
