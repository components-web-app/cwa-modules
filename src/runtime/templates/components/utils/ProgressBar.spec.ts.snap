// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProgressBar > snapshot > should match snapshot IF bar is NOT shown 1`] = `
<transition-stub
  appear="false"
  css="true"
  enteractiveclass="cwa-duration-200 cwa-ease-out"
  enterfromclass="cwa-transform cwa-opacity-0 cwa-w-0"
  entertoclass="cwa-opacity-100"
  leaveactiveclass="cwa-duration-300 cwa-ease-in cwa-delay-200"
  leavefromclass="cwa-opacity-100"
  leavetoclass="cwa-transform cwa-opacity-0"
  persisted="true"
>
  <div
    class="cwa-w-full cwa-h-0.5"
    style="display: none;"
  >
    <div
      class="cwa-bg-green-500 cwa-h-0.5 cwa-transition-all cwa-duration-300"
      style="width: 0%;"
    />
  </div>
</transition-stub>
`;

exports[`ProgressBar > snapshot > should match snapshot IF bar is shown with specified percentage 1`] = `
<transition-stub
  appear="false"
  css="true"
  enteractiveclass="cwa-duration-200 cwa-ease-out"
  enterfromclass="cwa-transform cwa-opacity-0 cwa-w-0"
  entertoclass="cwa-opacity-100"
  leaveactiveclass="cwa-duration-300 cwa-ease-in cwa-delay-200"
  leavefromclass="cwa-opacity-100"
  leavetoclass="cwa-transform cwa-opacity-0"
  persisted="true"
>
  <div
    class="cwa-w-full cwa-h-0.5"
  >
    <div
      class="cwa-bg-green-500 cwa-h-0.5 cwa-transition-all cwa-duration-300"
      style="width: 75%;"
    />
  </div>
</transition-stub>
`;
